.github/workflows/pytest.yml

```diff
@@ -43,6 +42,9 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
           cache: "pip"
+        # run full test matrix only on PRs to the master branch
+        if: github.ref == 'refs/heads/master' || matrix.runner == 'ubuntu-20.04' && matrix.python-version == '3.8'
```

Comment:
Isn't the idea to skip the job completely instead of only one of the steps?

Reply:
We still want to run the tests once for PRs to dev, with the latest version of ubuntu and python. And test will all the combinations for PRs to master.